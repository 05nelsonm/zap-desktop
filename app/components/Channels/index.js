export ChannelCapacity from './ChannelCapacity'
export ChannelMoreButton from './ChannelMoreButton'
export ChannelStatus from './ChannelStatus'
export ChannelSummaryList from './ChannelSummaryList'
export ChannelSummaryListItem from './ChannelSummaryListItem'
