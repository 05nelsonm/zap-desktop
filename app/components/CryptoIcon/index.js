import CryptoIcon from './CryptoIcon'

export default CryptoIcon
