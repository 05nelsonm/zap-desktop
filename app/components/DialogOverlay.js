import styled from 'styled-components'
import { Flex } from 'rebass'

const Overlay = styled(Flex)`
  position: absolute;
  z-index: 99999;
  background-color: rgba(255, 255, 255, 0.25);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`

export default Overlay
