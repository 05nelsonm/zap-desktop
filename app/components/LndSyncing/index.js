import LndSyncing from './LndSyncing'

export default LndSyncing
