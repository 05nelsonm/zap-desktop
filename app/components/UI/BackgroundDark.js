import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundDark
 * @example
 * <BackgroundDark />
 */
class BackgroundDark extends React.Component {
  render() {
    return <Box bg="darkestBackground" color="primaryText" {...this.props} />
  }
}

export default BackgroundDark
