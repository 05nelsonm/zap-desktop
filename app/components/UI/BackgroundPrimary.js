import React from 'react'
import { Box } from 'rebass'

/**
 * @render react
 * @name BackgroundPrimary
 * @example
 * <BackgroundPrimary />
 */
class BackgroundPrimary extends React.Component {
  render() {
    return <Box bg="primaryColor" color="primaryText" {...this.props} />
  }
}

export default BackgroundPrimary
