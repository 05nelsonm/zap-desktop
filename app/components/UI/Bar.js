import styled from 'styled-components'
import { variant, space, height } from 'styled-system'

const bars = variant({
  key: 'bars',
})

const Bar = styled.hr`
  height: 1px;
  margin: 0;
  padding: 0;
  border: 0;
  background-color: ${props => props.theme.colors.primaryText};
  ${space}
  ${height}
  ${bars}
`

Bar.defaultProps = {
  variant: 'normal',
}

Bar.displayName = 'Bar'

export default Bar
