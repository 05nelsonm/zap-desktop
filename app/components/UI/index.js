export ActionBar from './ActionBar'
export AmountInput from './AmountInput'
export BackgroundPrimary from './BackgroundPrimary'
export BackgroundSecondary from './BackgroundSecondary'
export BackgroundTertiary from './BackgroundTertiary'
export Bar from './Bar'
export Button from './Button'
export Checkbox from './Checkbox'
export Countdown from './Countdown'
export CryptoAmountInput from './CryptoAmountInput'
export CurrencyFieldGroup from './CurrencyFieldGroup'
export DataRow from './DataRow'
export Dialog from './Dialog'
export Dropdown from './Dropdown'
export FiatAmountInput from './FiatAmountInput'
export Form from './Form'
export GlobalStyle from './GlobalStyle'
export Header from './Header'
export Heading from './Heading'
export Input from './Input'
export Label from './Label'
export LightningInvoiceInput from './LightningInvoiceInput'
export Link from './Link'
export MainContent from './MainContent'
export Message from './Message'
export Modal from './Modal'
export Notification from './Notification'
export Page from './Page'
export Panel, { PanelHeader, PanelBody, PanelFooter } from './Panel'
export PasswordInput from './PasswordInput'
export QRCode from './QRCode'
export Radio from './Radio'
export RadioGroup from './RadioGroup'
export Range from './Range'
export Select from './Select'
export Sidebar from './Sidebar'
export Span from './Span'
export Spinner from './Spinner'
export StatusIndicator from './StatusIndicator'
export Text from './Text'
export TextArea from './TextArea'
export Titlebar from './Titlebar'
export Toggle from './Toggle'
export Value from './Value'
export Wizard from './Wizard'
