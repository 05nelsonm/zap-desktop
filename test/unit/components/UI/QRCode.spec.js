import React from 'react'
import renderer from 'react-test-renderer'
import { QRCode } from 'components/UI'

describe('component.UI.QRCode', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<QRCode value="qwerty" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
