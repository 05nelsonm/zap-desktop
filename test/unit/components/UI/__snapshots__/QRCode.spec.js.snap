// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.QRCode should render correctly 1`] = `
.c0 {
  position: relative;
  width: 180px;
  height: 180px;
  display: inline-block;
}

.c1 {
  position: relative;
  width: 180px;
  height: 180px;
}

.c2 {
  border: 1px solid;
  border-color: #fd9800;
  border-radius: 10%;
  position: absolute;
  top: 0;
  left: 0;
  display: inline-block;
  width: 180px;
  height: 180px;
  -webkit-clip-path: polygon(0 0,25% 0,25% 25%,0 25%);
  clip-path: polygon(0 0,25% 0,25% 25%,0 25%);
}

.c3 {
  border: 1px solid;
  border-color: #fd9800;
  border-radius: 10%;
  position: absolute;
  top: 0;
  right: 0;
  display: inline-block;
  width: 180px;
  height: 180px;
  -webkit-clip-path: polygon(75% 0,100% 0%,100% 25%,75% 25%);
  clip-path: polygon(75% 0,100% 0%,100% 25%,75% 25%);
}

.c4 {
  border: 1px solid;
  border-color: #fd9800;
  border-radius: 10%;
  position: absolute;
  bottom: 0;
  left: 0;
  display: inline-block;
  width: 180px;
  height: 180px;
  -webkit-clip-path: polygon(0 75%,25% 75%,25% 100%,0 100%);
  clip-path: polygon(0 75%,25% 75%,25% 100%,0 100%);
}

.c5 {
  border: 1px solid;
  border-color: #fd9800;
  border-radius: 10%;
  position: absolute;
  bottom: 0;
  right: 0;
  display: inline-block;
  width: 180px;
  height: 180px;
  -webkit-clip-path: polygon(75% 75%,100% 75%,100% 100%,75% 100%);
  clip-path: polygon(75% 75%,100% 75%,100% 100%,75% 100%);
}

.c6 {
  border-style: solid;
  border-color: white;
  border-width: 8px;
  position: absolute;
  top: calc(10% - 2px);
  left: calc(10% - 2px);
}

<div
  className="c0"
  size={150}
>
  <div
    className="c1"
    size={150}
  >
    <div
      className="c2"
      size={150}
    />
    <div
      className="c3"
      size={150}
    />
    <div
      className="c4"
      size={150}
    />
    <div
      className="c5"
      size={150}
    />
  </div>
  <svg
    bg="white"
    className="c6"
    color="black"
    height={150}
    shapeRendering="crispEdges"
    theme={
      Object {
        "buttons": Object {
          "normal": Object {
            "&:focus": Object {
              "backgroundColor": "#353745",
            },
            "&:hover:enabled": Object {
              "backgroundColor": "#353745",
            },
            "backgroundColor": "#313340",
            "color": "#fd9800",
          },
          "primary": Object {
            "backgroundColor": "#fd9800",
            "color": undefined,
          },
          "secondary": Object {
            "&.active": Object {
              "opacity": 1,
            },
            "&:focus": Object {
              "opacity": 1,
            },
            "&:hover:enabled": Object {
              "opacity": 1,
            },
            "backgroundColor": "transparent",
            "color": "#ffffff",
            "opacity": 0.6,
          },
        },
        "cards": Object {
          "error": Object {
            "backgroundColor": "#330d0d",
            "color": "#e63939",
          },
          "success": Object {
            "backgroundColor": "#0d331a",
            "color": "#39e673",
          },
          "warning": Object {
            "backgroundColor": "#4a2c00",
            "color": "#fd9800",
          },
        },
        "colors": Object {
          "gray": "#959595",
          "highlight": "#353745",
          "lightningBrown": "#4a2c00",
          "lightningOrange": "#fd9800",
          "mudBrown": "#330d0d",
          "pineGreen": "#0d331a",
          "primaryColor": "#242633",
          "primaryText": "#ffffff",
          "secondaryColor": "#373947",
          "superGreen": "#39e673",
          "superRed": "#e63939",
          "tertiaryColor": "#313340",
        },
        "fontSizes": Object {
          "0": "8px",
          "1": "10px",
          "2": "11px",
          "3": "13px",
          "4": "15px",
          "5": "18px",
          "6": "30px",
          "7": "60px",
          "l": "15px",
          "m": "13px",
          "s": "11px",
          "xl": "18px",
          "xs": "10px",
          "xxl": "30px",
          "xxs": "8px",
          "xxxl": "60px",
        },
        "fontWeights": Object {
          "light": 300,
          "normal": 400,
        },
        "fonts": Object {
          "0": "Roboto, system-ui, sans-serif",
          "sans": "Roboto, system-ui, sans-serif",
        },
        "letterSpacings": Object {
          "caps": "0.025em",
          "normal": "normal",
        },
        "messages": Object {
          "error": Object {
            "color": "#e63939",
          },
          "processing": Object {
            "color": "#fd9800",
          },
          "success": Object {
            "color": "#39e673",
          },
          "warning": Object {
            "color": "#fd9800",
          },
        },
        "name": "dark",
        "palette": Object {
          "black": "#000000",
          "deepseaBlue": "#242633",
          "gray": "#959595",
          "hoverSeaBlue": "#353745",
          "hoverSeaGray": "#f2f2f2",
          "lightGreen": "#eefff4",
          "lightOrange": "#fff3e1",
          "lightRed": "#ffeded",
          "lightningBrown": "#4a2c00",
          "lightningOrange": "#fd9800",
          "mudBrown": "#330d0d",
          "pineGreen": "#0d331a",
          "seaBlue": "#313340",
          "seaGray": "#f3f3f3",
          "superGreen": "#39e673",
          "superRed": "#e63939",
          "underwaterBlue": "#373947",
          "underwaterGray": "#ebebeb",
          "white": "#ffffff",
        },
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          45,
          72,
          108,
        ],
        "statuses": Object {
          "closing": Object {
            "color": "#e63939",
          },
          "offline": Object {
            "color": "#959595",
          },
          "online": Object {
            "color": "#39e673",
          },
          "pending": Object {
            "color": "#fd9800",
          },
        },
      }
    }
    viewBox="0 0 21 21"
    width={150}
  >
    <path
      d="M0,0 h21v21H0z"
      fill="white"
    />
    <path
      d="M0 0h7v1H0zM8 0h1v1H8zM12 0h1v1H12zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM8 1h2v1H8zM11 1h1v1H11zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h5v1H8zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h1v1H8zM10 3h1v1H10zM12 3h1v1H12zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM9 4h4v1H9zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM8 5h1v1H8zM10 5h1v1H10zM12 5h1v1H12zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM0 8h2v1H0zM4 8h3v1H4zM12 8h1v1H12zM15 8h1v1H15zM17,8 h4v1H17zM1 9h4v1H1zM8 9h1v1H8zM12 9h3v1H12zM17 9h2v1H17zM0 10h1v1H0zM4 10h3v1H4zM10 10h1v1H10zM12 10h2v1H12zM18 10h2v1H18zM1 11h1v1H1zM3 11h2v1H3zM7 11h6v1H7zM15 11h1v1H15zM19 11h1v1H19zM1 12h1v1H1zM3 12h5v1H3zM10 12h2v1H10zM14 12h3v1H14zM19,12 h2v1H19zM8 13h2v1H8zM11 13h2v1H11zM17 13h2v1H17zM0 14h7v1H0zM11 14h1v1H11zM14 14h3v1H14zM19 14h1v1H19zM0 15h1v1H0zM6 15h1v1H6zM8 15h2v1H8zM13 15h2v1H13zM16 15h1v1H16zM20,15 h1v1H20zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM8 16h1v1H8zM10 16h1v1H10zM12 16h3v1H12zM16 16h2v1H16zM20,16 h1v1H20zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM9 17h2v1H9zM12 17h5v1H12zM18,17 h3v1H18zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM12 18h2v1H12zM16 18h3v1H16zM0 19h1v1H0zM6 19h1v1H6zM8 19h5v1H8zM15 19h2v1H15zM0 20h7v1H0zM8 20h2v1H8zM11 20h1v1H11zM14 20h2v1H14zM17 20h1v1H17zM20,20 h1v1H20z"
      fill="black"
    />
  </svg>
</div>
`;
